# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleLoaderBase.h
# Commit: 0b3f0972e002
# Full Hash: 0b3f0972e002f6b010b4683fc7c8e4e366b2cf4e
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move DOM module loader into own file; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137531
# ==============================================================================

diff -r 98d6cb49f4fd -r 0b3f0972e002 dom/script/ModuleLoaderBase.h
--- a/dom/script/ModuleLoaderBase.h	Mon Feb 21 15:39:07 2022 +0000
+++ b/dom/script/ModuleLoaderBase.h	Mon Feb 21 15:39:08 2022 +0000
@@ -86,7 +86,9 @@
       mozilla::MaybeOneOf<JS::SourceText<char16_t>, JS::SourceText<Utf8Unit>>;
 
   // Methods that must be overwritten by an extending class
-  virtual void EnsureModuleHooksInitialized() = 0;
+  virtual void EnsureModuleHooksInitialized() {
+    MOZ_ASSERT(false, "You must override EnsureModuleHooksInitialized");
+  }
   virtual nsresult StartModuleLoad(ScriptLoadRequest* aRequest) = 0;
   virtual void ProcessLoadedModuleTree(ModuleLoadRequest* aRequest) = 0;
   virtual nsresult CompileOrFinishModuleScript(